batcher/aligned-batcher/src/lib.rs

```diff
@@ -1295,6 +1334,12 @@ impl Batcher {
             return Err(e);
```

Comment:
Just by reading the code (didn't tested this scenario) seems that we are doing a double flush on some entries.
This is because we are sending a `CreateNewTaskError` message to entries of the `finalized_batch`, and doing the same on entries of the queue inside the `flush_queue_and_clear_nonce_cache` function. 
This was right before, since no entries were shared between this two sets of entries, but now there are some entries repeated.

Reply:
Yes, proper flushing is handled in #1510.